tests/test_backends.py

```diff
@@ -434,3 +434,18 @@ def test_model_exist_check(cli_runner: CliRunner):
     assert (
         "test-model.gguf does not exist. Please download model first." in result.stdout
     )
+
+
+@pytest.mark.parametrize(
+    "flag, args_list, expected_flag, expected_value",
+    [
+        ("--foo", ["--foo", "2"], True, "2"),
+        ("--foo", ["--foo", "--foo"], True, None),
+        ("--foo", ["--foo=2"], True, "2"),
+        ("--foo", ["foo"], False, None),
```

Comment:
Nice test!

Addition Suggestions:
-Would be interesting to have the arg you are looking for not be the only item in the list
-Checking for an item in an empty list